### DISCUSSION TITLE
Secrets manager variables -  Invalid configuration

### DISCUSSION THREAD

**Discussion by ashley-jisc** [serverless/serverless#10823] [Q&A]
Hi, we have just upgraded to Serverless v3 and are experiencing the following warning on deployment:
```
Running "serverless" from node_modules

Warning: Invalid configuration encountered
  at 'provider.environment.VARIABLES_TEST': unsupported object format
```

AWS Secret:
`{
  "hello": "test"
}`

Serverless.yml
```
  environment:
    VARIABLES_TEST: ${ssm:aws/reference/secretsmanager/variablestest}
```

We can see the value listed if we run `sls print` though.

How can we resolve this warning?

**Comment by ashley-jisc** [comment]
Using `${ssm(raw):aws/reference/secretsmanager/variablestest}` appears to resolve the problem. But, why is this necessary for a simple key/value pair from Secrets Manager?

**Comment by pgrzesik** [comment]
Hello @ashley-jisc - in the first example, the JSON value was parsed and resulted in an object, which is an invalid value for `environment` variables. As you already shown, by using `ssm(raw)` it's possible to skip that JSON parsing.
